src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -57,6 +61,278 @@ static void RecordDistributedRelationDependencies(Oid distributedRelationId,
+
+	if (sizeList == NIL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
+						errmsg("cannot get the size")));
+	}
+
+	tableSizeStringInfo = (StringInfo) linitial(sizeList);
+	tableSizeString = tableSizeStringInfo->data;
+	relationSize = atol(tableSizeString);
+
+	return relationSize;
+}
+
+
+/*
+ * ShardIntervalsOnNode takes a WorkerNode then compares it with each placement
+ * of table. It returns shard intervals of table on that node as a list of shard
+ * intervals. Note that, shard intervals returned as elements of the list are
+ * not the copies but the pointers.
+ *
+ * DO NOT modify the shard intervals returned by this function.
+ */
+static List *
+ShardIntervalsOnWorkerNode(WorkerNode *workerNode, Oid relationId)
+{
+	DistTableCacheEntry *distTableCacheEntry = DistributedTableCacheEntry(relationId);
+	char *workerNodeName = workerNode->workerName;
+	uint32 workerNodePort = workerNode->workerPort;
+	List *shardIntervalList = NIL;
+	int shardIndex = 0;
+	int shardIntervalArrayLength = distTableCacheEntry->shardIntervalArrayLength;
+
+	for (shardIndex = 0; shardIndex < shardIntervalArrayLength; shardIndex++)
+	{
+		ShardPlacement *placementArray =
+			distTableCacheEntry->arrayOfPlacementArrays[shardIndex];
+		int numberOfPlacements =
+			distTableCacheEntry->arrayOfPlacementArrayLengths[shardIndex];
+		int placementIndex = 0;
+
+		for (placementIndex = 0; placementIndex < numberOfPlacements; placementIndex++)
+		{
+			ShardPlacement *placement = &placementArray[placementIndex];
+			char *shardNodeName = placement->nodeName;
+			uint32 shardNodePort = placement->nodePort;
+			uint64 shardId = placement->shardId;
+			bool metadataLock = false;
+
+			metadataLock = TryLockShardDistributionMetadata(shardId, ShareLock);
+
+			/* if the lock is not acquired warn the user */
+			if (metadataLock == false)
+			{
+				ereport(WARNING, (errcode(ERRCODE_LOCK_NOT_AVAILABLE),
+								  errmsg("lock is not acquired, size of shard %ld "
+										 "will be ignored", shardId)));
+				continue;
+			}
+
+			if (strcmp(shardNodeName, workerNodeName) == 0 &&
+				shardNodePort == workerNodePort)
+			{
+				ShardInterval *shardInterval =
+					distTableCacheEntry->sortedShardIntervalArray[shardIndex];
+				shardIntervalList = lappend(shardIntervalList, shardInterval);
+			}
+		}
+	}
+
+	return shardIntervalList;
+}
+
+
+/*
+ * GenerateSizeQueryOnMultiplePlacements generates a select size query to get
+ * size of multiple tables from the relation with distributedRelationId. Note
+ * that, different size functions supported by PG are also supported by this
+ * function changing the size query given as the last parameter to function.
+ * Format of sizeQuery is pg_*_size(%s). Examples of it can be found in the
+ * master_protocol.h
+ */
+static StringInfo
+GenerateSizeQueryOnMultiplePlacements(Oid distributedRelationId, List *shardIntervalList,
+									  char *sizeQuery)
+{
+	/* schema name will be used to form shard qualified name */
```

Comment:
Minor. We can delete this comment from the first line of the function? We don't generally put comments as the first line and this comment seems like nice to have :)